cmd/tsbs_generate_data/main.go

```diff
@@ -174,14 +178,31 @@ func main() {
 	}
 
 	rand.Seed(seed)
-	out := bufio.NewWriterSize(os.Stdout, inputBufSize)
+
+	// Prepare output file/STDOUT
+	var out  *bufio.Writer
+	var file io.Writer
+	var err  error
```

Comment:
only `var file io.Writer` is needed here, you only use `err` in the `if` branch so you can use `:=` there, and `out` is assigned afterward so you can use `:=` there too

Reply:
The same as above - extract to separate function